addons/mrp/models/mrp_production.py

```diff
@@ -725,6 +725,9 @@ def _check_production_lines(self):
                     raise ValidationError(_("The component %s should not be the same as the product to produce.") % production.product_id.display_name)
 
     def write(self, vals):
+        if 'move_byproduct_ids' in vals and 'move_finished_ids' not in vals:
+            vals['move_finished_ids'] = vals.get('move_finished_ids', []) + vals['move_byproduct_ids']
```

Comment:
If condition looks for `'move_finished_ids' not in vals`, then won't `vals.get('move_finished_ids', [])` always be `[]`?